ResearchKit/Common/ORKReviewStep.h

```diff
@@ -0,0 +1,85 @@
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import <ResearchKit/ORKStep.h>
+#import <ResearchKit/ORKResult.h>
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ The `ORKReviewStep` class is a concrete subclass of `ORKStep` that represents
+ a step in which existing question results can be reviewed by the user.
+ 
+ There are two separate scenarios for using `ORKReviewStep`. The first one is to embed a review step within an ongoing survey, which means an instance of `ORKReviewStep` might be placed at the end or in the middle of a task. The second scenario is meant to use `ORKReviewStep` standalone for reviewing an already completed task.
+ 
+ To use a review step, instantiate an `ORKReviewStep` object, fill in its properties, and include it in a task. Next, create a task view controller for the task and present it.
+ 
+ When a task view controller presents an `ORKReviewStep` object, it instantiates an `ORKReviewStepViewController` object to present the step. This view controller lists both steps and step results to provide all entered data at a glance. It also allows the user to navigate directly to a certain step. However, results may only be changed from there, if the review step is used within an ongoing survey (`embedded`).
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKReviewStep : ORKStep
+
+/**
+ Returns a new standalone review step that includes the specified identifier, steps, and result source.
+ 
+ @param identifier    The identifier of the step (a step identifier should be unique within the task).
+ @param steps         The steps that should be reviewed.
+ @param resultSource  The source that should be consulted to obtain the corresponding step results.
+ */
++ (instancetype)standaloneReviewStepWithIdentifier:(NSString *)identifier
+                                             steps:(nonnull NSArray *)steps
+                                      resultSource:(nullable id<ORKTaskResultSource, NSSecureCoding>)resultSource;
```

Comment:
Current it is possible to initialize a standalone review step with just steps and no result source. Is that an intended use case? I feel like they both should be non null. Also remove the `nonnull` identifier since we already did `NS_ASSUME_NONULL_BEGIN` at the top of the file.